### ISSUE TITLE
Issue importing obspy.signal

### ISSUE THREAD

**Issue by gbalme** [obspy/obspy#1140]
As told in issue #1137:

I'm working on windows with Anaconda Python 2.7 and Spyder, and I have a problem when importing obspy.signal module version 0.10.2,(same problem with version 0.10.1 and 0.10.0)

```python
File "C:\Anaconda\lib\site-packages\obspy\signal__init__.py", line 198, in 
from .spectral_estimation import PPSD

File "C:\Anaconda\lib\site-packages\obspy\signal\spectral_estimation.py", line 48, in 
from obspy.signal.invsim import cosine_taper

File "C:\Anaconda\lib\site-packages\obspy\signal\invsim.py", line 37, in 
from obspy.signal import util

File "C:\Anaconda\lib\site-packages\obspy\signal\util.py", line 25, in 
from obspy.signal.headers import clibsignal

File "C:\Anaconda\lib\site-packages\obspy\signal\headers.py", line 18, in 
clibsignal = _load_CDLL("signal")

File "C:\Anaconda\lib\site-packages\obspy\core\util\libnames.py", line 96, in _load_CDLL
msg = 'Could not load shared library "%s".\n\n %s' % (libname, str(e))

UnicodeDecodeError: 'ascii' codec can't decode byte 0xe9 in position 24: ordinal not in range(128)
```

I tried the 2 lines of code provided by megies in issue #1137, but a new error arises on the 2nd line provided:

```python
KeyError: (0, 0)
```

Futhermore, In C:\Anaconda\lib\site-packages\obspy\lib, there is only 1 file: "init.py"
And inside this file, just 2 lines:
```python
from future import (absolute_import, division, print_function, unicode_literals)
from future.builtins import * # NOQA
```

Also, on the line:
```python
from future.builtins import * # NOQA
```

I get a warning "unable to detect undefined names"

Also, if I just do "import obspy" , instead of "import obspy.signal", I don't get the unicodedecode error.
However, when I do for example:
```python
hilb = obspy.signal.filter.envelope(data)
```
Then There is another error arising:
```python
  File "C:\Anaconda\lib\site-packages\spyderlib\widgets\externalshell\sitecustomize.py", line 685, in runfile
    execfile(filename, namespace)

  File "C:\Anaconda\lib\site-packages\spyderlib\widgets\externalshell\sitecustomize.py", line 71, in execfile
    exec(compile(scripttext, filename, 'exec'), glob, loc)

  File "C:/Users/Zenbook/Documents/Thèse_GBalme/filtre_python.py", line 48, in <module>

  File "C:\Anaconda\lib\site-packages\obspy\core\util\deprecation_helpers.py", line 61, in __getattr__
    raise AttributeError

AttributeError
```

What's wrong ?
Thank you

**Comment by megies** [comment]
I think the problem is the empty `C:\Anaconda\lib\site-packages\obspy\lib` folder (the two-line `__init__.py` is OK), it's lacking the compiled C extensions.
How did you install ObsPy? Via `conda install -c obspy obspy` (i.e. precompiled binary package)?

**Comment by krischer** [comment]
The raised first error occurs because there is a non-ASCII char in the exception message which we attempt to reraise...I guess its the path of your script but we should really work around that.

You have a grave accent on the pathname of your script. On Python 2 that might screw things up - in particular on windows. I know its a huge problem if python on windows is installed in a path that has spaces in its name...

Another funny thing is the `sitecustomize.py` script that Spyder runs. What is it doing?

Can you try to run your script outside of Spyder and in a path that only contains ASCII chars?

**Comment by megies** [comment]
I've checked one windows pre-compiled file at anaconda (`win-64_obspy-0.10.2-np18py27_0.tar.bz2`) and it does contain the compiled C extensions in `Lib/site-packages/obspy-0.10.2-py2.7-win-amd64.egg/obspy/lib/`:
```
__init__.py
__init__.pyc
libevresp_Windows_64bit_py27.pyd
libgse2_Windows_64bit_py27.pyd
libmseed_Windows_64bit_py27.pyd
libsegy_Windows_64bit_py27.pyd
libsignal_Windows_64bit_py27.pyd
libtau_Windows_64bit_py27.pyd
```
We still need to know the means of installation you used, but I suspect that you did compile it yourself and something went wrong there..

**Comment by ThomasLecocq** [comment]
As @krischer said, it's most probably the "Thèse" in your path that is problematic... try to run your code from another folder, or rename it without accents...

**Comment by megies** [comment]
..still think this is only a follow up problem with the Exception printing. Underlying is a broken install I think.

**Comment by megies** [comment]
No feedback in two weeks, so closing. Feel free to reopen if problem persists.

**Comment by lileipku00** [comment]
I met this problem on Windows. There is no space or non-ascii chars in python path.

```
$ import obspy
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-8-6960ca51d676> in <module>()
----> 1 import obspy

D:\Anaconda2\lib\site-packages\obspy\__init__.py in <module>()
    179 from obspy.core.util.base import make_format_plugin_table
    180 read.__doc__ = \
--> 181     read.__doc__ % make_format_plugin_table("waveform", "read", numspaces=4)
    182 read_events.__doc__ = \
    183     read_events.__doc__ % make_format_plugin_table("event", "read",

D:\Anaconda2\lib\site-packages\obspy\core\util\base.pyc in make_format_plugin_table(group, method, numspaces, unindent_first_line)
    520         module_short = ":mod:`%s`" % ".".join(ep.module_name.split(".")[:3])
    521         func = load_entry_point(ep.dist.key,
--> 522                                 "obspy.plugin.%s.%s" % (group, name), method)
    523         func_str = ':func:`%s`' % ".".join((ep.module_name, func.__name__))
    524         mod_list.append((name, module_short, func_str))

D:\Anaconda2\lib\site-packages\pkg_resources\__init__.pyc in load_entry_point(dist, group, name)
    568 def load_entry_point(dist, group, name):
    569     """Return `name` entry point of `group` for `dist` or raise ImportError"""
--> 570     return get_distribution(dist).load_entry_point(group, name)
    571
    572

D:\Anaconda2\lib\site-packages\pkg_resources\__init__.pyc in load_entry_point(self, group, name)
   2749         if ep is None:
   2750             raise ImportError("Entry point %r not found" % ((group, name),))
-> 2751         return ep.load()
   2752
   2753     def get_entry_map(self, group=None):

D:\Anaconda2\lib\site-packages\pkg_resources\__init__.pyc in load(self, require, *args, **kwargs)
   2403         if require:
   2404             self.require(*args, **kwargs)
-> 2405         return self.resolve()
   2406
   2407     def resolve(self):

D:\Anaconda2\lib\site-packages\pkg_resources\__init__.pyc in resolve(self)
   2409         Resolve the entry point from its module and attrs.
   2410         """
-> 2411         module = __import__(self.module_name, fromlist=['__name__'], level=0)
   2412         try:
   2413             return functools.reduce(getattr, self.attrs, module)

D:\Anaconda2\lib\site-packages\obspy\io\mseed\core.py in <module>()
     20 from obspy.core.util.deprecation_helpers import (
     21     DynamicAttributeImportRerouteModule, ObsPyDeprecationWarning)
---> 22 from . import util
     23 from .headers import (DATATYPES, ENCODINGS, HPTERROR, HPTMODULUS, SAMPLETYPE,
     24                       SEED_CONTROL_HEADERS, UNSUPPORTED_ENCODINGS,

D:\Anaconda2\lib\site-packages\obspy\io\mseed\util.py in <module>()
     22 from obspy.core.util import score_at_percentile
     23 from obspy.core.util.decorator import deprecated
---> 24 from .headers import (ENCODINGS, ENDIAN, FIXED_HEADER_ACTIVITY_FLAGS,
     25                       FIXED_HEADER_DATA_QUAL_FLAGS,
     26                       FIXED_HEADER_IO_CLOCK_FLAGS, FRAME, HPTMODULUS,

D:\Anaconda2\lib\site-packages\obspy\io\mseed\headers.py in <module>()
     20
     21 # Import shared libmseed
---> 22 clibmseed = _load_cdll("mseed")
     23
     24

D:\Anaconda2\lib\site-packages\obspy\core\util\libnames.pyc in _load_cdll(name)
     95     except Exception as e:
     96         msg = 'Could not load shared library "%s".\n\n %s' % (libname, str(e))
---> 97         raise ImportError(msg)
     98     return cdll
     99

ImportError: Could not load shared library "libmseed_Windows_64bit_py27.pyd".

 LoadLibrary() argument 1 must be string, not unicode
```

**Comment by trichter** [comment]
@lileipku00 Thanks for reporting. This is a completely different error.  I'll open a new issue.

> Why unicode is preferred all through the obspy codes?

Because of the changed string model in Py3 unicode is also the standard string type in Py2 for obspy.

**Comment by megies** [comment]
@lileipku00 looks like this was fixed in #1636 and released in ObsPy 1.0.3. Can you please state your obspy  version (and soon, we want to release 1.1.0 within a day or two).

**Comment by megies** [comment]
Ah.. didn't want to post in here.. locking this ticket to avoid more cross postings..
